### PR TITLE
[BACKLOG-41048] Added code changes for excluding anonymous role from BA roles in a VFS connection create/edit page.

### PR DESCRIPTION
**Pull request by nawaz34-hitachivantara**
Jira - https://hv-eng.atlassian.net/browse/BACKLOG-41048

### CODE REVIEW

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java [hunk 1]
```diff
@@ -233,11 +234,11 @@ public RoleListWrapper getRoles( @QueryParam( "addExtraRoles" ) String addExtraR
   }
 
   /**
-   * Get a list of the all roles in the platform including Anonymous and Administrator.
+   * Get a list of the all roles in the platform including Administrator.If excludeAnonymous is false, Anonymous role is included, else excluded.
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java:237]: Missing a space after the first period.
Also, the `@param` doclet is missing.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java [hunk 2]
```diff
@@ -112,6 +114,19 @@ public RoleListWrapper getSystemRoles() {
...
+  public RoleListWrapper getAllRoles( boolean excludeAnonymous ) {
+    if ( !excludeAnonymous ) {
+      return getAllRoles();
+    }
+
+    IPentahoSession session = PentahoSessionHolder.getSession();
+    String anonymousRole = PentahoSystem.get( String.class, "singleTenantAnonymousAuthorityName", session );
+    List<String> roles = getAllRoles().getRoles();
+    roles.remove( anonymousRole );
+    return new RoleListWrapper( roles );
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java:126]: You should instead follow the pattern used for the admin role and receive the anonymous role as a property specified by the resource class. Then, before removing check it's specified (not `null`).

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java [hunk 3]
```diff
@@ -112,6 +114,19 @@ public RoleListWrapper getSystemRoles() {
     return new RoleListWrapper( systemRoles );
   }
 
+  public RoleListWrapper getAllRoles( boolean excludeAnonymous ) {
+    if ( !excludeAnonymous ) {
+      return getAllRoles();
+    }
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java:120]: Please, do this the other way around, by changing `getAllRoles()` to call this one. Will allow you not have to create a new role list wrapper.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1564**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/UserRoleListResourceTest.java [hunk 4]
```diff
@@ -126,14 +126,25 @@ public void testGetRoles() throws Exception {
...
-    RoleListWrapper testWrapper = userRoleListResource.getAllRoles();
+    RoleListWrapper testWrapper = userRoleListResource.getAllRoles( false );
     assertEquals( mockWrapper, testWrapper );
 
-    verify( userRoleListServiceMock, times( 1 ) ).getAllRoles();
+    verify( userRoleListServiceMock, times( 1 ) ).getAllRoles( false );
+  }
+
+  @Test
+  public void testGetAllRolesWithoutAnonymous() {
```
Comment (dcleao) [inline extensions/src/test/java/org/pentaho/platform/web/http/api/resources/UserRoleListResourceTest.java:140]: There's also a `UserRoleListServiceTest` class where we should add tests for the new functionality.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 27m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 21m 41s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java (hunk 1 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java]: Addressed

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java (hunk 2 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java]: Addressed

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java (hunk 3 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java]: Addressed

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/UserRoleListResourceTest.java (hunk 4 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/test/java/org/pentaho/platform/web/http/api/resources/UserRoleListResourceTest.java:140]: Addressed

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 28m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 28m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java [hunk 5]
```diff
@@ -107,6 +101,7 @@ private static UserRoleListService createUserRoleListServiceFromConfiguration()
   @NonNull
   private static UserRoleListService createUserRoleListService( @NonNull List<String> systemRoles,
                                                                 @NonNull String adminRole,
+                                                                @NonNull String anonymousRole,
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java:110]: Doclet not updated.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java [hunk 6]
```diff
@@ -30,17 +30,12 @@
 import org.pentaho.platform.engine.security.DefaultUsernameComparator;
 import org.pentaho.platform.web.http.api.resources.services.UserRoleListService;
 
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.*;
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java]: Do not use * imports.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java [hunk 7]
```diff
@@ -30,17 +30,12 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
-import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
-import static javax.ws.rs.core.MediaType.APPLICATION_XML;
-import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static javax.ws.rs.core.MediaType.*;
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java]: Do not use * imports.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java [hunk 8]
```diff
@@ -104,14 +110,15 @@ public RoleListWrapper getAllRoles() {
     }
 
     existingRoles.addAll( allRoles );
-
+    if ( excludeAnonymous && getAnonymousRole() != null ) {
+      existingRoles.remove( getAnonymousRole() );
+    }
```
Comment (dcleao) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java:115]: Please surround this new block with an empty line, before and after.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — CHANGES_REQUESTED** [review]

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java [hunk 9]
```diff
@@ -166,7 +165,7 @@ public void testDoGetAllRoles() {
     doReturn( roles ).when( userRoleListService1 ).getAllRoles();
     doReturn( extraRoles ).when( userRoleListService ).getExtraRoles();
 
-    RoleListWrapper roleListWrapper = userRoleListService.getAllRoles();
+    RoleListWrapper roleListWrapper = userRoleListService.getAllRoles( false );
```
Comment (dcleao) [inline extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java:168]: Should still call `getAllRoles()`.

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java [hunk 10]
```diff
@@ -30,8 +30,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (dcleao) [inline extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java:33]: Do not use star imports.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java (hunk 5 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleListResource.java:111]: Added the doclet

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java (hunk 8 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java:116]: Addressed

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java (hunk 9 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java]: Addressed

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java (hunk 10 above)
Comment (nawaz34-hitachivantara) [reply extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListServiceTest.java]: Addressed

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 38m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 21m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1565**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5640/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**